core-blocks/image/edit.js

```diff
@@ -172,7 +172,7 @@ class ImageEdit extends Component {
 
 	updateAlignment( nextAlign ) {
 		const extraUpdatedAttributes = [ 'wide', 'full' ].indexOf( nextAlign ) !== -1 ?
-			{ width: undefined, height: undefined } :
+			{ width: undefined, height: undefined, url: this.props.image ? this.props.image.source_url : this.props.attributes.url } :
```

Comment:
Honestly, I'm not certain this behavior is what we want as some people might still want to use an alternative size (it could be big enough) while using wide/full aligned images. cc @jasmussen

Reply:
Can you elaborate on what this means?

As far as I'm concerned, if an image is set to _wide_ or _fullwide_, we defer completely to the theme to define what that means.

Reply:
The `url` used in the `<img>` tag is chosen using an inspector control. If we want to always use the "full sized" image for these, than it's fine, we can keep this change, if we want to allow users to choose which "image size" (including custom registered sizes which can be bign enough), than we shouldn't reset.

Reply:
Ah, gotcha. So a user uploads a 4000*4000px photo of a badger, then wants to make it _wide_, but they feel 4000*4000 might be too much resolution and chooses a smaller size even if it's wide? Sure, that makes sense to me.

Reply:
Ok so I took this for a spin and it looks like we're hiding the "Image Size" selector when in full/wide alignments. This means even if you want to change those you can't. which means:

 1- We can ship this PR as: always full image in wide alignedd images
 2- Update the PR to always show the image size selector

I'd appreciate an opinion quickly because I'm trying to get this in for 3.5 :)